### PR TITLE
[FSDP2] Simplified `_move_states_to_device`

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #122908
* #119302
* __->__ #122907

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/_composable/fsdp/fully_shard.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def fully_shard(
 
     managed_modules = _get_managed_modules(module)
     params, buffers = _get_managed_states(managed_modules)
-    _move_states_to_device(params, buffers, device, mesh_info)
+    _move_states_to_device(params, buffers, device)
```
Comment (awgu) [inline torch/distributed/_composable/fsdp/fully_shard.py:116]: unused argument

File: torch/distributed/_composable/fsdp/_fsdp_init.py [hunk 2]
```diff
@@ -133,7 +132,7 @@ def _move_states_to_device(
             # Keep meta-device tensors on meta device for deferred init
             continue
         if isinstance(tensor, DTensor):
-            if (dtensor_mesh_type := tensor._spec.mesh.device_type) != device.type:
+            if (dtensor_mesh_type := tensor.device_mesh.device_type) != device.type:
```
Comment (awgu) [inline torch/distributed/_composable/fsdp/_fsdp_init.py:135]: use public API

**Review by Skylion007 — APPROVED** [review]

**Comment by awgu** [comment]
It looks like this PR already merged, but the PR did not get closed 🤔

**Pull request closed without merging**